src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.PendingBranchesCollection.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class AbstractFlowPass<TLocalState, TLocalFunctionState>
+    {
+        protected sealed class PendingBranchesCollection : IEnumerable<PendingBranch>
+        {
+            private ArrayBuilder<PendingBranch> _branches;
+            private PooledDictionary<LabelSymbol, ArrayBuilder<PendingBranch>>? _branchesToLabel;
+
+            internal PendingBranchesCollection()
+            {
+                _branches = ArrayBuilder<PendingBranch>.GetInstance();
+            }
+
+            internal void Free()
+            {
+                _branches.Free();
+                _branches = null!;
+                FreeBranchesToLabel();
+            }
+
+            internal void Clear()
+            {
+                _branches.Clear();
+                FreeBranchesToLabel();
```

Comment:
>FreeBranchesToLabel [](http://example.com/codeflow?start=16&length=19)

Is it not possible to clear the dictionary?

Reply:
Yes, there is a `Clear()` but we'd still need to free all `ArrayBuilder<PendingBranch>` instances, and then we'd also have to consider an empty dictionary as a base case similar to a `null` dictionary. Will leave as is since the current code seems more straightforward.